plugins/org.wso2.developerstudio.eclipse.artifact.synapse.api/src/org/wso2/developerstudio/eclipse/artifact/synapse/api/ui/wizard/SynapseAPICreationWizard.java

```diff
@@ -144,36 +144,40 @@ protected boolean isRequiredWorkingSet() {
 	}
 	
 	private String getTemplateContent(){
-		String content = new String();
+		StringBuilder content = new StringBuilder();
 		/*
 		 * FIXME: use template extension-point instead of hard-coding template
 		 * content
 		 */
-		content += "<api xmlns=\"";
-		content += "http://ws.apache.org/ns/synapse\"";
-		content += " context=\"";
-		content += artifactModel.getContext();
-		content += "\" name=\"";
-		content += artifactModel.getName();
-		content += "\""; 
-		if(artifactModel.getHostname()!=null && artifactModel.getHostname().length()>0){
-			content += " hostname=\"";
-			content += artifactModel.getHostname();
-			content += "\"";
+		content.append("<api xmlns=\"");
+		content.append("http://ws.apache.org/ns/synapse\"");
+		content.append(" context=\"");
+		if (!artifactModel.getContext().startsWith("/")) {
+			content.append("/").append (artifactModel.getContext());
+		} else {
+			content.append(artifactModel.getContext());
 		}
-		if(artifactModel.getPort()>0){
-			content += " port=\"";
-			content += artifactModel.getPort();
-			content += "\"";
+		content.append("\" name=\"");
+		content.append(artifactModel.getName());
+		content.append("\"");
+		if (artifactModel.getHostname() != null && artifactModel.getHostname().length() > 0) {
+			content.append(" hostname=\"");
+			content.append(artifactModel.getHostname());
+			content.append("\"");
 		}
-		content += ">\n";
-		content += "<resource methods=\"GET\">";
-		content += "<inSequence/>";
-		content += "<outSequence/>";
-		content += "<faultSequence/>";
-		content += "</resource>\n";
-		content += "</api>";
-		return content;
+		if (artifactModel.getPort() > 0) {
+			content.append(" port=\"");
+			content.append(artifactModel.getPort());
+			content.append("\"");
+		}
+		content.append(">\n");
+		content.append("<resource methods=\"GET\">");
+		content.append("<inSequence/>");
+		content.append("<outSequence/>");
+		content.append("<faultSequence/>");
+		content.append("</resource>\n");
+		content.append("</api>");
```

Comment:
These all appends can be merged to one with one big string

Reply:
Done